reviewing/list/src/pruneTree.ts

```diff
@@ -0,0 +1,78 @@
+import { type DependenciesHierarchy, type PackageNode } from '@pnpm/reviewing.dependencies-hierarchy'
+import { type PackageDependencyHierarchy } from './types'
+import { createHash } from 'crypto'
+
+export function pruneTreeToGetFirst10EndLeafs (trees: PackageDependencyHierarchy[] | null): PackageDependencyHierarchy[] {
+  if (trees === null) {
+    return []
+  }
+
+  return trees.map((tree) => {
+    const endLeafPaths: PackageNode[][] = []
+    const visitedNodes = new Set<string>()
+
+    function dfs (node: PackageNode, path: PackageNode[]): void {
```

Comment:
what does dfs mean? Use a descriptive name for the function.

Reply:
True, I renamed it as `findEndLeafs`.